### ISSUE TITLE
"Generic.Arrays.ArrayIndent" conflict with "Generic.WhiteSpace.ScopeIndent", and "PEAR.WhiteSpace.ScopeIndent", in arrays with short syntax.

### ISSUE THREAD

**Issue by aboalnaga** [squizlabs/PHP_CodeSniffer#2281]
When using short array syntax like this
```php
$shortArray = [
  "hello",
"world"
];
```
using the configuration below
```xml
<rule ref="Generic.Arrays.ArrayIndent"/>
<rule ref="Generic.WhiteSpace.ScopeIndent">
  <properties>
    <property name="exact" value="true"/>
  </properties>
</rule>
<rule ref="PEAR.WhiteSpace.ScopeIndent">
  <properties>
    <property name="exact" value="true"/>
  </properties>
</rule>
```
Output shows conflicting errors as follows
```
---------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 2 LINES
---------------------------------------------------------------------------------
 2 | ERROR | [x] Array key not indented correctly; expected 4 spaces but found 0
   |       |     (Generic.Arrays.ArrayIndent.KeyIncorrect)
 3 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
   |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 3 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
   |       |     (PEAR.WhiteSpace.ScopeIndent.IncorrectExact)
---------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------
```
> Note: this issue doesn't occur for long syntax; as both "Generic.WhiteSpace.ScopeIndent", and "PEAR.WhiteSpace.ScopeIndent" don't report any of indentations as an error.

**Comment by gsherwood** [comment]
This is caused by your use of the `exact` property, which should almost never be used. What that does is tell the scope indent sniff to never allow any custom indentation rules.

There isn't really anything I can do here as the content of arrays needs to follow the same rules as elsewhere (consider an array where the values are anon functions) and I can't hard code array indentation rules into the scope indent sniff.

I strongly recommend not using that exact setting for the scope indent sniff and trying to use another sniff to enforce whatever rules you are using the exact flag to enforce.

**Comment by aboalnaga** [comment]
I knew that removing `exact` would make this work, but I need to flag excess indentations as well, for `phpcbf` to correctly format indentations, as I seldom receive correct indented code.

I attempted to modify the sniff by escaping the array tokens, but couldn't reinstate it inside anonymous functions, with no knowledge of how and little time on my hands, the `exact` was gone.

You've mentioned 

> use another sniff to enforce whatever rules you are using the exact flag to enforce

Have you got some rule in mind that will flag excess indentations?

**Comment by gsherwood** [comment]
I think I've got a solution for this where I can ignore indentation inside arrays without breaking the indent checks of closures (and similar constructs) that might be inside them. Will try some things and report back.

**Comment by gsherwood** [comment]
I've pushed up a change (will be in 3.4.1) that ignores exact indentation checking inside all arrays (both short and long). On 3.4.1, you'll now see these errors with the ruleset and sample code you posted:
```
FILE: temp.php
--------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------------
 3 | ERROR | [x] Array key not indented correctly; expected 4 spaces but found 2
   |       |     (Generic.Arrays.ArrayIndent.KeyIncorrect)
 4 | ERROR | [x] Array key not indented correctly; expected 4 spaces but found 0
   |       |     (Generic.Arrays.ArrayIndent.KeyIncorrect)
--------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------

```
